src/Authentication/AuthenticationManager.php

```diff
@@ -109,7 +109,7 @@ public function getAccount(RequestInterface $request, $cache = TRUE) {
         drupal_page_is_cacheable(variable_get('restful_page_cache', FALSE));
 
         // User didn't authenticate against any provider, so we throw an error.
-        throw new UnauthorizedException('Bad credentials');
+        throw new UnauthorizedException('Anonymous user resolved for a resource that requires authentication.');
```

Comment:
"Bad credentials" should remain there (possible with some more explanation) - since this is telling us credentials were sent, but not correct. It's not just trying to access a resource as anon.